### PR TITLE
Newfood Atomization: Grown Plants & Unit Tests

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/54768
- https://github.com/tgstation/tgstation/pull/55040
- https://github.com/tgstation/tgstation/pull/57232
- https://github.com/tgstation/tgstation/pull/58795
- https://github.com/tgstation/tgstation/pull/71765

## About The Pull Request

Converts all grown food into newfood

Adds unit tests for checking all food edibility and ensuring food have icons

Also adds precaution so crafted food does not inherit light effects from a ingredient (not especially an issue now, but it absolutely will be when customfood element is added)

Additionally, establishes a framework for junkiness. Its only modified by snackcode, which is not yet converted, so this will be unused for the time being.

EDIT: I've fixed incorrect microwave paths from the meat pr, and added IS_EDIBLE checks to fix processors, drying racks, IV drips, & others not accepting food

TODO

- [x] Plant Genes inherit correctly 
- [x] Potency affects reagent amount
- [x] Dry racks work as intended

## Why It's Good For The Game

Newfood atomization

Unit tests ensure no regression, and that all food works as intended

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/4bd6b974-eff6-4285-9eba-829bd4eba167

drying rack

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/5ecd8837-e500-4290-95c1-972d9bacbe31

</details>

## Changelog
:cl: rkz, qustinnus, Timberpoes, ShizCalev, cacogen, WeSoda25
tweak: chaplains enjoy the taste of Holy Melons, and get a mood event for it.
add: converts grown food to newfood
remove: temporarily removes plants from customfood
code: unit tests to check food edibility & icons
code: establishes junkiness framework for snacks
fix: fixes incorrect microwave cooking paths for some meats
fix: makes processors, microwaves, drying racks accept all food again
/:cl:

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -1023,6 +1023,11 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 /obj/item/proc/on_mob_death(mob/living/L, gibbed)
 
+/obj/item/proc/add_juice()
+	if(reagents)
+		return 1
+	return 0
```
Comment (PowerfulBacon) [inline code/game/objects/items.dm]: This seems incredibly weird, what does the return value mean and what does this proc actually do? If this is a stub for subtype implementation then shouldn't it always return false to indicate that its not actually doing anything?

Comment (Tsar-Salat) [reply code/game/objects/items.dm]: My bad, I moved the add_juice proc down from /obj/item/grown/ during testing.

I've moved it back up. Newfood doesnt really affect that system. I just forgot to fix it. Thanks for reminding me

Comment (Tsar-Salat) [reply code/game/objects/items.dm]: reverted in b0ca606

**Comment by itsmeow** [comment]
I am once again asking for an UpdatePaths...

**Comment by Tsar-Salat** [comment]
> I am once again asking for an UpdatePaths...

image

😏

**Review by itsmeow — APPROVED** [review]

**Comment by itsmeow** [comment]
deconflict with meat pr

**Comment by Tsar-Salat** [comment]
g2g

**Review by itsmeow — APPROVED** [review]

**Comment by Rukofamicom** [comment]
This PR made an unnecessary variable name change across the board.

`materials` did not need to be renamed to `custom_materials`

**Pull request merged**